packages/cuba-react/src/ui/table/DataTable.tsx

```diff
@@ -49,12 +52,27 @@ export interface DataTableProps<E> extends MainStoreInjected {
   canSelectRowByClick?: boolean,
   buttons?: JSX.Element[],
   tableProps?: TableProps<E>,
+  /**
+   * @deprecated use `columnDefinitions` instead. If used together, `columnDefinitions` will take precedence.
+   */
   columnProps?: ColumnProps<E>,
+  /**
+   * You need to use either `columnDefinitions` or `fields` for the `DataTable` to work
+   */
+  columnDefinitions?: Array<string | ColumnDefinition<E>>
+}
+
+export interface ColumnDefinition<E> {
+  field?: string,
+  columnProps: ColumnProps<E>
 }
 
 @injectMainStore
 @observer
 export class DataTable<E> extends React.Component<DataTableProps<E>> {
+
+  static readonly NO_COLUMN_DEF_ERROR = 'You need to provide either `columnDefinitions` or `fields` prop to the `DataTable`';
```

Comment:
When we throw this error, is it possible to add info about context for ease of debug. E.g. entity name or may be something else about current DataTable?

Reply:
done